File: 백준/Silver/10828. 스택/스택.py
# 10828 스택

'''
push X: 정수 X를 스택에 넣는 연산이다.
pop: 스택에서 가장 위에 있는 정수를 빼고, 그 수를 출력한다. 만약 스택에 들어있는 정수가 없는 경우에는 -1을 출력한다.
size: 스택에 들어있는 정수의 개수를 출력한다.
empty: 스택이 비어있으면 1, 아니면 0을 출력한다.
top: 스택의 가장 위에 있는 정수를 출력한다. 만약 스택에 들어있는 정수가 없는 경우에는 -1을 출력한다.
'''

from collections import deque
import sys

N = int(sys.stdin.readline())
dq = deque()
len = 0
for i in range(N):
    order = sys.stdin.readline().split()

    if order[0] == 'push':
        dq.append(order[1])
        len += 1

    elif order[0] == 'pop':
        if len != 0:
            print(dq.pop())
            len -= 1
        else:
            print(-1)

    elif order[0] == 'size':
        print(len)

    elif order[0] == 'empty':
        if len != 0:
            print(0)
        else:
            print(1)

    else:
        if len != 0:
            print(dq[-1])
        else:
            print(-1)

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);

	int N;
	cin >> N;
	stack<int>s;
	for (int i = 0;i < N;i++) {
		string order;
		cin >> order;
		if (order == "push") {
			int X;
			cin >> X;
			s.push(X);
		}
		else if (order == "pop") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.top() << '\n';
				s.pop();
			}
		}
		else if (order =="size") {
			cout << s.size() << '\n';
		}
		else if (order == "empty") {
			if (s.empty()) {
				cout << 1 << '\n';
			}
			else {
				cout << 0 << '\n';
			}
		}
		else {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.top() << '\n';
			}
		}
	}


	return 0;
}